src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerOptions.cs

```diff
@@ -63,5 +63,29 @@ public ConsoleLoggerFormat Format
         /// </summary>
         [System.ObsoleteAttribute("ConsoleLoggerOptions.UseUtcTimestamp has been deprecated. Use ConsoleFormatterOptions.UseUtcTimestamp instead.")]
         public bool UseUtcTimestamp { get; set; }
+
+        /// <summary>
+        /// Gets or sets the desired console logger behavior when buffer becomes full. Defaults to <c>Wait</c>.
+        /// </summary>
+        public ConsoleLoggerBufferFullMode BufferFullMode { get; set; }
+
+        private int _maxQueuedMessages = 1048576;
+
+        /// <summary>
+        /// Gets or sets the maximum number of enqueued messages. Defaults to 1048576.
+        /// </summary>
+        public int MaxQueueLength
+        {
+            get => _maxQueuedMessages;
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentException(nameof(MaxQueueLength));
```

Comment:
>MaxQueueLength [](http://example.com/codeflow?start=55&length=14)

do we use the property name? looking at other places in this file and seeing we are using `nameof(value)`

Reply:
I think so. `ArgumentOutOfRangeException` is also probably slightly better.